receiver/signalfxreceiver/receiver.go

```diff
@@ -90,27 +92,30 @@ type sfxReceiver struct {
 var _ component.MetricsReceiver = (*sfxReceiver)(nil)
 
 // New creates the SignalFx receiver with the given configuration.
-func New(
+func newReceiver(
 	logger *zap.Logger,
 	config Config,
-	nextConsumer consumer.MetricsConsumer,
-) (component.MetricsReceiver, error) {
-
-	if nextConsumer == nil {
-		return nil, errNilNextConsumer
+) *sfxReceiver {
+	r := &sfxReceiver{
+		logger: logger,
+		config: &config,
 	}
 
-	if config.Endpoint == "" {
-		return nil, errEmptyEndpoint
-	}
+	return r
+}
 
-	r := &sfxReceiver{
-		logger:       logger,
-		config:       &config,
-		nextConsumer: nextConsumer,
-	}
+func (r *sfxReceiver) RegisterMetricsConsumer(mc consumer.MetricsConsumer) {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
Is this function ever called concurrently? The lock doesn't hurt, but I wonder if this ever is needed just to be sure I understand everything correctly.

Reply:
The lock also helps if the create*Receiver methods are called in separate goroutines, even if not concurrently, to establish memory coherency.  I think from looking at the service core code that they are not, but I would rather assume as little as possible in the receiver unless it is something documented.

That being said, I would also synchronize the use of the global map in the factory as well to be consistent.